app/helpers/advanced_messenger_helper.rb

```diff
@@ -47,4 +47,34 @@ def getUnreadNotificationsForCurrentUserGroupByIssues()
     def getUnreadNotificationsForCurrentUserCount()
         return Journal.where("read_by_users ILIKE ?", '%"' + User.current.id.to_s + '":{"read":0%').count;
     end
+
+    def getUsersReadStatus(journal)
+        read_by_users = JSON.parse(journal.read_by_users)
+        read_statuses = Hash.new
+        users = Hash.new
+        read_by_users.keys.each_with_index do |user_id, index|
+            if (!read_by_users[user_id]["read"]) 
+                # no read status for this user i.e. the note is not of interest for him
+                next
+            end
+            user = User.find(user_id)
```

Comment:
I see that you make a query to db for each user for a note for every notes, even if the user is the same.

More precisely, if we have

```
note1: user1, user2
note2: user1, user2
note3: user1, user2
```

you will make 3 (notes) x 2 (users) = 6 database access when only 2 where necessary.

Is this correct? If so, can you cache the result?

Reply:
You are right. thanks. I modified this